documents/write_acls.go

```diff
@@ -355,3 +355,70 @@ func (cd *CoreDocument) GetTransitionRule(ruleID []byte) (*coredocumentpb.Transi
 
 	return nil, ErrTransitionRuleMissing
 }
+
+// isRoleAssignedToRules checks if the given roleID is used in any transition rules except default rules
+func isRoleAssignedToRules(cd *CoreDocument, roleID []byte) bool {
+	fieldMap := defaultRuleFieldProps()
+	for _, rule := range cd.Document.TransitionRules {
+		// check if the rule is the default rule
+		if _, ok := fieldMap[hexutil.Encode(rule.Field)]; ok {
+			continue
+		}
+
+		// check if the role exists in the rule
+		if byteutils.ContainsBytesInSlice(rule.Roles, roleID) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// deleteRoleFromDefaultRules deletes the role from the default rules.
+func deleteRoleFromDefaultRules(cd *CoreDocument, roleID []byte) {
```

Comment:
Why not having CoreDocument as the pointer receiver in this scenario, compared to the `deleteRule` function below?